backend/prisma/schema.prisma

```diff
@@ -17,17 +17,27 @@ enum UserType {
   ADMIN // admins are also teachers
 }
 
+enum AuthenticationProvider {
+  MICROSOFT
+}
+
 model User {
-  id    Int      @id @default(autoincrement())
-  name  String?
-  email String   @unique
-  type  UserType
+  id                     Int                    @id @default(autoincrement())
+  name                   String?
+  oidcSub                String?
```

Comment:
as pointed out elsewhere, this should be something like `externalId` if we want to make it independent of the auth provider.